churaverse-plugins-client/src/churarenPlugin/churarenPlayerPlugin/churarenPlayerPlugin.ts

```diff
@@ -0,0 +1,294 @@
+    if (this.playerPluginStore.ownPlayerId === player.id) {
+      this.ghostModeIndicatorUi?.ghostModeIcon.activate()
+      this.hideItemBox()
+    }
+    this.changeGhostPlayer(player.id)
+    this.updateGhostPlayerList()
+  }
+
+  private readonly onInvicibleTime = (ev: InvicibleTimeEvent): void => {
+    const playerId = ev.id
+    const invicibleTime = ev.invicibleTime
+    this.playerPluginStore.playerRenderers.get(playerId)?.blinkPlayer(invicibleTime)
+  }
+
+  private readonly getItem = (ev: GetChurarenItemEvent): void => {
+    const item = this.itemPluginStore.items.get(ev.itemId)
+    const renderer = this.itemPluginStore.itemRenderers.get(ev.itemId)
+
+    // TODO: 錬金アイテムを持っている場合も考慮する
+    if (item === undefined || renderer === undefined) return
+
+    this.playerItemStore.materialItems.set(ev.playerId, item)
+    this.playerItemStore.materialItemRenderers.set(ev.itemId, renderer)
+    this.itemPluginStore.items.delete(ev.itemId)
+    this.itemPluginStore.itemRenderers.delete(ev.itemId)
+    if (ev.playerId === this.playerPluginStore.ownPlayerId) {
+      this.updateMaterialItemBox(ev.playerId)
+    }
+  }
+
+  private readonly onPlayerWalk = (ev: PlayerWalkEvent): void => {
+    const player = this.playerPluginStore.players.get(ev.id)
+    if (player === undefined) return
+    const speed = ev.speed ?? GRID_SIZE / GRID_WALK_DURATION_MS
+
+    // TODO: 錬金アイテムを持っている場合の処理を追加
+    const itemBoxes = this.playerItemStore.materialItems.getAllItem(ev.id)
+    itemBoxes.forEach((item: Item, index: number) => {
+      if (item === undefined) return
+      const renderer = this.playerItemStore.materialItemRenderers.get(item.itemId)
+      const dest = player.position.copy()
+      dest.x -= player.direction.x * (40 + index * 40)
+      dest.y -= player.direction.y * (40 + index * 40)
+      this.setupChase(renderer, item, dest, speed)
+    })
+  }
+
+  private readonly dropItem = (ev: DropChurarenItemEvent): void => {
+    const renderer = this.playerItemStore.materialItemRenderers.get(ev.itemId)
+    if (renderer == null) return
+    renderer.destroy()
+    this.playerItemStore.materialItemRenderers.delete(ev.itemId)
+    this.playerItemStore.materialItems.delete(ev.playerId, ev.itemId)
+
+    if (ev.playerId === this.playerPluginStore.ownPlayerId) {
+      this.updateMaterialItemBox(ev.playerId)
+      const dropItemData: DropChurarenItemData = { playerId: ev.playerId, itemId: ev.itemId }
+      const dropItemMessage = new DropChurarenItemMessage(dropItemData)
+      this.networkPluginStore.messageSender.send(dropItemMessage)
+    }
+  }
+
+  private readonly logPlayerDeathByBoss = (ev: LivingDamageEvent): void => {
+    // TODO: プレイヤーがボスによって死亡した時のログを流す
+    //  (churarenBossPluginのChurarenEnemyDamageCauseかどうかを判定する)
+  }
+
+  private readonly onChangePlayerName = (ev: PlayerNameChangeEvent): void => {
+    if (!this.isActive) return
+    this.updateGhostPlayerList()
+  }
+
+  private setupChase(
+    renderer: any,
```

Comment:
anyを使っているのは型安全でないのでよろしくないですねー

`IItemRenderer`として受け取れば良いかなと思いますがうまくいきませんか？
```suggestion
    renderer: IItemRenderer,
```

そもそもとして、`renderer.chase()`を呼び出しているだけなのであれば関数に切り分ける必要もないかなと思います